### PR TITLE
Remove bootstrap style import `@import 'bootstrap/scss/forms'`

### PR DESCRIPTION
**Pull request by gitstart-sourcegraph**
## Problem Statement
We depend on Bootstrap styles in our global styles a lot. But we don't plan to move forward with Bootstrap because we're working on our design system called Wildcard. To ease the transition to our styles and gain more control over visual changes, we want to move Bootstrap styles that we need into our codebase and eventually drop Bootstrap dependency.

## Ref
[Parent Issue](https://github.com/sourcegraph/sourcegraph/issues/29200)
[SG Issue](https://github.com/sourcegraph/sourcegraph/issues/30279)
[Gitstart Task](https://app.gitstart.com/clients/sourcegraph/tickets/SG-30279)

## Preview App
[Storybook](https://client-sourcegraph-storybook-pr-219.onrender.com/)
[Frontend](https://client-sourcegraph-frontend-pr-219.onrender.com/)

## Success Criteria
- [ ] All highlighted imports has been removed
- [ ] All css rules currently used from the imports are now available in style sheets

## Implementation Details
Let's also remove the following imports and copy CSS rules that we need to our CSS files:

1) remove the imports @import 'bootstrap/scss/forms from global index.scss file
2) copy styles used in codebase into the global forms.scss file

## Test plan

Ensure that UI doesn't have regressions.

### CODE REVIEW

**Comment by gitstart-sourcegraph** [comment]
@valerybugakov,
kindly help review spec PR. 
Also, we're currently unable to view updated chromatic review on CI, any ideas why?

**Comment by valerybugakov** [comment]
@gitstart-sourcegraph, we disabled Chromatic check for draft PRs. If you need to check a Chromatic diff after making several changes — mark your PR as ready for review and convert it back to draft. We will look into implementing commit message hints to trigger this step for specific commits later.

**Comment by gitstart-sourcegraph** [comment]
@valerybugakov, PR ready for review. kindly take a look

**Review by valerybugakov — CHANGES_REQUESTED** [review]
I noticed some undesired UI changes happening because we define form rules earlier, and they have higher priority than some Bootstrap files like `_custom-forms.scss`.

Screen Shot 2022-02-14 at 09 25 57

Compare it to the UI on the main branch or our dogfood instance: https://k8s.sgdev.org/users/admin/settings/emails. Let's double-check other usages of the updated styles and ensure that we don't have unexpected changes.

File: .github/workflows/label-move.yml [hunk 1]
```diff
@@ -198,25 +198,3 @@ jobs:
               }
             }
           }' -f project=$PROJECT_ID -f node_id=$NODE_ID
-  wcag-accessibility-board:
```
Comment (valerybugakov) [inline .github/workflows/label-move.yml:201]: Redundant changes?

File: client/branded/src/global-styles/forms.scss [hunk 2]
```diff
@@ -1,3 +1,29 @@
+@import 'wildcard/src/global-styles/breakpoints';
+
+:root {
+    --font-size-base: $font-size-base;
```
Comment (valerybugakov) [inline client/branded/src/global-styles/forms.scss:4]: This CSS variable can be used in other parts of global styles. Let's define it in `client/branded/src/global-styles/index.scss`.

File: client/branded/src/global-styles/forms.scss [hunk 3]
```diff
@@ -1,3 +1,29 @@
...
+    --input-font-size-sm: 0.786em;
```
Comment (valerybugakov) [inline client/branded/src/global-styles/forms.scss:5]: In Bootstrap this variable is defined relative to the `--font-size-base` variable. Let's preserve this relation:

```scss
--font-size-base: $font-size-base;
--input-font-size-sm: calc(var(--font-size-base) * 0.875);
```

This comment is applicable for other CSS variables defined here. Let's double-check if they should be calculated from already existing variables.

File: client/branded/src/global-styles/forms.scss [hunk 4]
```diff
@@ -1,3 +1,29 @@
...
+    --input-line-height: 1.429;
```
Comment (valerybugakov) [inline client/branded/src/global-styles/forms.scss:6]: Should rely on `$line-height-base` define in `client/branded/src/global-styles/index.scss` as Bootstrap does it.

File: client/branded/src/global-styles/forms.scss [hunk 5]
```diff
@@ -1,3 +1,29 @@
...
+    --input-padding-x-sm: 0.5rem;
+    --input-height-sm: calc(1.5em + (var(--input-padding-y-sm) * 2) + var(--input-height-border));
+    --input-font-size: var(--font-size-base);
+    --input-height-inner: calc(1.429em + (var(--input-padding-y) * 2));
+    --form-group-margin-bottom: 1rem;
+    --form-text-margin-top: 0.25rem;
+    --form-check-input-gutter: 1.25rem;
+    --form-check-input-margin-x: 0.25rem;
+    --input-height-inner-half: calc(0.714em + var(--input-padding-y));
+    --border-radius: 3px;
```
Comment (valerybugakov) [inline client/branded/src/global-styles/forms.scss:24]: It's already defined in `client/branded/src/global-styles/border-radius.scss`

File: client/branded/src/global-styles/forms.scss [hunk 6]
```diff
@@ -1,3 +1,29 @@
...
+    --font-size-base: $font-size-base;
+    --input-font-size-sm: 0.786em;
+    --input-line-height: 1.429;
+    --input-height-border: 2px;
+    --input-padding-y: 0.375rem;
+    --input-padding-x: 0.75rem;
+    --input-height-inner-quater: calc(0.35714em + 0.1875rem);
+    --input-height: calc(1.429em + (var(--input-padding-y) * 2) + var(--input-height-border));
+    --input-font-weight: 400;
+    --input-border-width: 1px;
```
Comment (valerybugakov) [inline client/branded/src/global-styles/forms.scss:13]: This variable [is defined](https://github.com/sourcegraph/sourcegraph/pull/30352/files#diff-b0ec73ef001a2ee6c05a4c91bc55c4f37c9d04cb57c990e90c8019cf2c9f6f04R28) in a related PR. Let's make sure that we do not duplicate them.

File: client/branded/src/global-styles/forms.scss [hunk 7]
```diff
@@ -1,3 +1,29 @@
+@import 'wildcard/src/global-styles/breakpoints';
+
+:root {
+    --font-size-base: $font-size-base;
+    --input-font-size-sm: 0.786em;
+    --input-line-height: 1.429;
+    --input-height-border: 2px;
+    --input-padding-y: 0.375rem;
+    --input-padding-x: 0.75rem;
```
Comment (valerybugakov) [inline client/branded/src/global-styles/forms.scss:9]: Defined in another PR too. Let's avoid duplication.

**Review by valerybugakov — CHANGES_REQUESTED** [review]

File: client/branded/src/global-styles/forms.scss (hunk 3 above)
Comment (gitstart-sourcegraph) [reply client/branded/src/global-styles/forms.scss]: We replaced the value of `$input-font-size-sm` in `typography.scss` so couldn't use this suggestion, kindly take a look

File: .github/workflows/label-move.yml (hunk 1 above)
Comment (gitstart-sourcegraph) [reply .github/workflows/label-move.yml]: Reverted now.

File: client/branded/src/global-styles/typography.scss [hunk 8]
```diff
@@ -87,6 +87,10 @@ $input-font-size-sm: (11/14) + em;
     line-height: 1rem;
 }
 
+:root {
+    --input-font-size-sm: #{$input-font-size-sm};
+}
```
Comment (valerybugakov) [inline client/branded/src/global-styles/typography.scss:92]: Let's move all global CSS variables affected by this PR into a `global-styles/variables.scss` and import it in `global-styles/index.scss` in a follow-up PR. It's confusing to see global CSS variables related to input in multiple places.

```scss
$input-font-size-sm: (11/14) + em;
// ...other SASS variables defined in `global-styles/index.scss`

:root {
    --spacer: #{$spacer};
    --font-size-base: #{$font-size-base};
    --line-height-base: #{$line-height-base};

  // input
  --input-font-size-sm: #{$input-font-size-sm};
  // ...other input related CSS variables
}

**Review by valerybugakov — APPROVED** [review]

File: client/branded/src/global-styles/typography.scss (hunk 8 above)
Comment (gitstart-sourcegraph) [reply client/branded/src/global-styles/typography.scss:92]: To be resolved in #32117

**Pull request merged**